### DISCUSSION TITLE
Encounter an error (CPU : Function not authorized for current protection level)

### DISCUSSION THREAD

**Discussion by M1chaelWang** [gijzelaerr/python-snap7#540] [Q&A] [answered]
When I try to call this function
`db2_data = client.upload(2)`
The program says Error: b'CPU : Function not authorized for current protection level'

I tried to use `client.plc_stop()` but it didn't work. THX!! :)

**Comment by lupaulus** [comment]
@M1chaelWang Working on a PLC 1200/1500 ?

**Reply by lupaulus** [reply]
[See S7 1200/1500 Notes](https://github.com/fbarresi/Sharp7?tab=readme-ov-file)

**Reply by lupaulus** [reply]
![image](https://github.com/user-attachments/assets/c68894d3-877c-465d-bd44-84e0575acdd1)

**Reply by M1chaelWang** [reply]
Well, actually I am using snap7.server for mimicking a siemens 7 server, so it's a fake server instead of a real one :)

**Reply by M1chaelWang** [reply]
<S7CpuInfo ModuleTypeName: b'CPU 315-2 PN/DP' SerialNumber: b'S C-C2UR28922012' ASName: b'SNAP7-SERVER' Copyright: b'Original Siemens Equipment' ModuleName: b'CPU 315-2 PN/DP'>

So it's a S7-300 Cpu 315-2 Pn/Dp.

**Reply by lupaulus** [reply]
![image](https://github.com/user-attachments/assets/1c89d046-2ecc-40ac-b1a1-4d1c02a4bdf5)

**Reply by lupaulus** [reply, marked as answer]
@M1chaelWang Not implemented in the base library, not sure if you can use a SIEMENS PLC-SIM Software

**Reply by M1chaelWang** [reply]
Thanks for your reply!
